node-notes-cli/app.js

```diff
@@ -0,0 +1,63 @@
+const fs = require('fs');
+const command = process.argv[2];
+const textString = process.argv[3];
+const idNum = parseInt(textString);
+const newNote = process.argv[4];
+
+const read = () => {
+  fs.readFile('data.json', 'utf8', (err, jsonData) => {
+    if (err) throw err;
+    const data = JSON.parse(jsonData).notes;
+    for (const [key, value] of Object.entries(data)) {
+      console.log(`${key}: ${value}`);
+    }
+  });
+};
+
+const create = textString => {
+  fs.readFile('data.json', 'utf8', (err, jsonData) => {
+    if (err) throw err;
+    const dataObj = JSON.parse(jsonData);
+    dataObj.notes[dataObj.nextId] = textString;
+    dataObj.nextId++;
+    updateNotes(dataObj);
+  });
+};
+
+const deleteNote = indexNum => {
+  fs.readFile('data.json', 'utf8', (err, jsonData) => {
+    if (err) throw err;
+    const dataObj = JSON.parse(jsonData);
+    delete dataObj.notes[indexNum];
+    updateNotes(dataObj);
+  });
+};
+
+const changeNote = (indexNum, newNote) => {
+  fs.readFile('data.json', 'utf8', (err, jsonData) => {
+    if (err) throw err;
+    const dataObj = JSON.parse(jsonData);
+    if (indexNum < dataObj.nextId) {
+      dataObj.notes[indexNum] = newNote;
+      updateNotes(dataObj);
+    } else {
```

Comment:
The update operation should not be able to create new notes, including re-using previously-deleted note ids.